app/javascript/components/region-form/index.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from 'patternfly-react';
+import createSchema from './region-form.schema';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import { API } from '../../http_api';
+import miqRedirectBack from "../../helpers/miq-redirect-back";
+
+const RegionForm = ({ id, maxDescLen }) => {
+  const [description, setInitialValues] = useState({});
+
+  useEffect(() => {
+    miqSparkleOn();
+    API.get(`/api/regions/${id}?attributes=description`)
+      .then(({ description }) => setInitialValues({ description }))
+      .then(miqSparkleOff);
+  }, [id]);
+
+  const onCancel = () => {
+    const message = __('Edit of Region was cancelled by the user');
+    miqRedirectBack(message, 'success', `/ops/explorer/?button=cancel`);
+  };
+
+  const onSubmit = (values) => {
+    miqSparkleOn();
+    const message = __('Region was saved');
+    const submitUrl = `/ops/explorer/${id}?button=save`;
+
+    return API.post(`/api/regions/${id}`, {
+        action: 'edit',
+        resource: {
+          description: values.description,
+        }
+      })
+      .then(() => miqRedirectBack(message, 'success', submitUrl))
+      .catch(miqSparkleOff);
+  };
+
+  if (!description) {
+    return null;
+  }
+
+  return (
+    <Grid fluid style={{ paddingTop: 20 }}>
+      <MiqFormRenderer
+        initialValues={description}
```

Comment:
```suggestion
        initialValues={{ description }}
```

Reply:
not sure why we need those extra curlies here, those populate text field with value `[object Object]` instead of an actual value of the description.

Reply:
It's about semantics, the `initialValues` should contain the `description` not be it. The extra curly here basically does:
```
initialValues = { description: description }
```